[Win] Retry failed VirtualAlloc commit

iangrunert opened the pull request:
#### 47afd2e05464b9cb5d9afe1617b2a6ce10c9ffc2
<pre>
[Win] Retry failed VirtualAlloc commit
<a href="https://bugs.webkit.org/show_bug.cgi?id=296938">https://bugs.webkit.org/show_bug.cgi?id=296938</a>

Reviewed by Yusuke Suzuki.

Address some crashes in Layout Tests due to failed VirtualAlloc calls to
MEM_COMMIT. If you commit too quickly, sometimes Windows will fail calls
as it hasn&apos;t grown the page table (ERROR_COMMITMENT_LIMIT); waiting and
trying again can help prevent a crash here.

Prior art from Firefox:

<a href="https://hacks.mozilla.org/2022/11/improving-firefox-stability-with-this-one-weird-trick/">https://hacks.mozilla.org/2022/11/improving-firefox-stability-with-this-one-weird-trick/</a>
Canonical link: <a href="https://commits.webkit.org/298793@main">https://commits.webkit.org/298793@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e68de799e24638e2a9c5b76886df9c4c5c29a25c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116586 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36247 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26812 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122658 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/67253 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36953 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44835 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88547 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42980 "Unexpected infrastructure issue, retrying build") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119535 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29478 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104599 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69007 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28553 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22704 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66325 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/108700 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98859 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22864 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125796 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115114 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43485 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32666 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97221 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43849 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100801 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97013 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24718 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42332 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20264 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39447 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43368 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/143812 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42837 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37024 "Found 1 new JSC stress test failure: wasm.yaml/wasm/v8/import-function.js.wasm-collect-continuously (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46177 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44539 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3b676e1488122462e34bcd513b8487062f342cc5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48995

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6418be1f1e4d20cb1aa34ba4a8fb227d760ed706. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48995

Jarred-Sumner commented:
Enabling libpas on Windows doubled the number of crashes in Bun, many of which included libpas in the stack trace. We disabled libpas in Bun v1.2.18 for now. I wonder if this fixes that.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e68de799e24638e2a9c5b76886df9c4c5c29a25c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48995

iangrunert committed:
[Win] Retry failed VirtualAlloc commit
https://bugs.webkit.org/show_bug.cgi?id=296938

Reviewed by Yusuke Suzuki.

Address some crashes in Layout Tests due to failed VirtualAlloc calls to
MEM_COMMIT. If you commit too quickly, sometimes Windows will fail calls
as it hasn't grown the page table (ERROR_COMMITMENT_LIMIT); waiting and
trying again can help prevent a crash here.

Prior art from Firefox:

https://hacks.mozilla.org/2022/11/improving-firefox-stability-with-this-one-weird-trick/
Canonical link: https://commits.webkit.org/298793@main
